### PR TITLE
Liquid staking unit and integration tests

### PR DESCRIPTION
**Pull request by sasurobert**
Liquid staking unit tests
Liquid staking integration tests

### CODE REVIEW

File: integrationTests/testProcessorNode.go [hunk 1]
```diff
@@ -2275,7 +2343,7 @@ func (tpn *TestProcessorNode) LoadTxSignSkBytes(skBytes []byte) {
 // ProposeBlock proposes a new block
 func (tpn *TestProcessorNode) ProposeBlock(round uint64, nonce uint64) (data.BodyHandler, data.HeaderHandler, [][]byte) {
 	startTime := time.Now()
-	maxTime := time.Second * 2
+	maxTime := time.Second * 200000
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/testProcessorNode.go:2346]: done

File: vm/systemSmartContracts/delegation_test.go [hunk 2]
```diff
@@ -1786,9 +1786,16 @@ func TestDelegationSystemSC_ExecuteUnDelegateUserErrorsWhenGettingMinimumDelegat
...
+		MinDeposit:          big.NewInt(50),
+		MinDelegationAmount: big.NewInt(50),
+	}
+	marshaledData, _ := d.marshalizer.Marshal(managementData)
+	eei.SetStorageForAddress(d.delegationMgrSCAddress, []byte(delegationManagementKey), marshaledData)
+
 	output := d.Execute(vmInput)
 	assert.Equal(t, vmcommon.UserError, output)
-	assert.True(t, strings.Contains(eei.returnMessage, "error getting minimum delegation amount"))
+	assert.True(t, strings.Contains(eei.returnMessage, "invalid value to undelegate - need to undelegate all - do not leave dust behind"))
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vm/systemSmartContracts/delegation_test.go:1798]: yes. it is. Error getting minimum amount does not happen at all. Only if delegation management was not instantiated.

File: vm/systemSmartContracts/delegation.go [hunk 3]
```diff
@@ -1602,19 +1602,13 @@ func (d *delegation) unDelegateValueFromAddress(
...
-		return vmcommon.UserError
-	}
-
-	minDelegationAmount := delegationManagement.MinDelegationAmount
-
 	remainedFund := big.NewInt(0).Sub(activeFund.Value, valueToUnDelegate)
-	if remainedFund.Cmp(zero) > 0 && remainedFund.Cmp(minDelegationAmount) < 0 {
+	err = d.checkRemainingFundValue(remainedFund)
+	if err != nil {
 		d.eei.AddReturnMessage("invalid value to undelegate - need to undelegate all - do not leave dust behind")
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vm/systemSmartContracts/delegation.go:1608]: no, this has to be left as it is for backward compatibility purposes.

File: integrationTests/vm/delegation/liquidStaking_test.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	idxProposers := make([]int, numOfShards+1)
+	for i := 0; i < numOfShards; i++ {
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	var tokenID []byte
+	for _, node := range nodes {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/vm/delegation/liquidStaking_test.go:122]: would leave as it is.

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**